File: src/main/java/org/gangel/graphomance/arangodb/ArangoConnection.java
package org.gangel.graphomance.arangodb;

import com.arangodb.ArangoDB;
import lombok.Builder;
import lombok.Getter;
import org.gangel.graphomance.Connection;

@Builder
public class ArangoConnection implements Connection {

    @Getter
    private ArangoDB db;

    @Getter
    private ArangoConnectionSettings connectionSettings;

    @Override
    public void open() {
    }

    @Override
    public void close() {
        db.shutdown();
    }
}

File: src/main/kotlin/org/gangel/graphomance/arangodb/ArangoConnection.kt
package org.gangel.graphomance.arangodb

import com.arangodb.ArangoDB
import org.gangel.graphomance.Connection

class ArangoConnection(
	val db: ArangoDB,
	val connectionSettings: ArangoConnectionSettings
) : Connection {
	override fun open() {}
	override fun close() {
		db.shutdown()
	}
}
